### ISSUE TITLE
Define serving tags in signature_def for ConveRT

### ISSUE THREAD

**Issue by vjtrost88** [PolyAI-LDN/polyai-models#26]
My team is very excited to use your model, but our workflow consists of taking all these new pre-trained NLU models that come out and serving them with Tensorflow Serving so we can hit them for embeddings via an API (we're still trying to see which ones are best for our use cases).

The ConveRT models weren't saved with the default serving tags which is what hinders them from giving us embeddings when we serve them. In digging around the TF docs, I found the `saved_model_cli` which allows you to see the `signature_def` for models:
```
~/models/conveRT/1$ saved_model_cli show --dir . --tag_set serve --signature_def serving_default
Traceback (most recent call last):
  File "/home/vince/Env/python3/bin/saved_model_cli", line 8, in <module>
    sys.exit(main())
  File "/home/vince/Env/python3/lib/python3.6/site-packages/tensorflow_core/python/tools/saved_model_cli.py", line 990, in main
    args.func(args)
  File "/home/vince/Env/python3/lib/python3.6/site-packages/tensorflow_core/python/tools/saved_model_cli.py", line 702, in show
    _show_inputs_outputs(args.dir, args.tag_set, args.signature_def)
  File "/home/vince/Env/python3/lib/python3.6/site-packages/tensorflow_core/python/tools/saved_model_cli.py", line 137, in _show_inputs_outputs
    tag_set)
  File "/home/vince/Env/python3/lib/python3.6/site-packages/tensorflow_core/python/tools/saved_model_utils.py", line 120, in get_meta_graph_def
    ' could not be found in SavedModel')
RuntimeError: MetaGraphDef associated with tag-set serve could not be found in SavedModel
```
And just as an example of what I see when I try this on another model (universal sentence encoder):
```
~/models/use/1$ saved_model_cli show --dir . --tag_set serve --signature_def serving_default
The given SavedModel SignatureDef contains the following input(s):
  inputs['inputs'] tensor_info:
      dtype: DT_STRING
      shape: (-1)
      name: Placeholder:0
The given SavedModel SignatureDef contains the following output(s):
  outputs['outputs'] tensor_info:
      dtype: DT_FLOAT
      shape: (-1, 512)
      name: module_apply_default/Encoder_en/hidden_layers/l2_normalize:0
Method name is: tensorflow/serving/predict
```

My question is how can I add these default serving tags? Or would it be possible for you to support this in your models that are saved on TF-Hub? I've been wrestling with this for a week so I figured I'd just ask. Thanks in advance for any help here.

**Comment by matthen** [comment]
Hi Vince, thanks for the interest!

I think you could add a signature def with any name following something like this: https://stackoverflow.com/a/51440571

so maybe:

```python
import tensorflow as tf
import tensorflow_hub as hub
import tensorflow_text
from tensorflow.saved_model import simple_save

export_dir = "new_saved_model"

with tf.Session(graph=tf.Graph()) as sess:
    module = hub.Module("http://models.poly-ai.com/convert/v1/model.tar.gz") 
    text_input = tf.placeholder(name='text', dtype=tf.string, shape=[None])
    sess.run([tf.global_variables_initializer(), tf.tables_initializer()])
    encodings = module(text_input)

    simple_save(
        sess,
        export_dir,
        inputs={'text': text_input},
        outputs={'embeddings': embeddings},
        legacy_init_op=tf.tables_initializer())
```

(not tested)

Alternatively when making requests to the tensorflow serving server, could you specify the "default" signature rather than letting it default to "serving_default", which doesn't exist?

I think the next issue you would run into is ensuring the ops from tensorflow text are loaded on the tensorflow serving server. It looks like these are included in version 2.0.0: https://github.com/tensorflow/serving/releases/tag/2.0.0 `Some Tensorflow Text ops have been added to ModelServer (specifically constrained_sequence_op, sentence_breaking_ops, unicode_script_tokenizer, whitespace_tokenizer, wordpiece_tokenizer)`

**Comment by vjtrost88** [comment]
This worked! We haven't switched to TF 2 yet so I'll cross that bridge when I get there. Thanks again!
